src/tree/localProject/ConfigGroupTreeItem.ts

```diff
@@ -3,37 +3,101 @@
+                                        void ext.ui.showWarningMessage(localize('mustContainLocs', `"{0}" must include "api_location" and "app_location". See the [workflow file guide](https://aka.ms/AAbrcox).`, yamlFile));
+                                        continue;
+                                    }
+
+                                    buildConfigs.set('api_location', step['with']['api_location'])
+                                    buildConfigs.set('app_location', step['with']['app_location'])
+
+                                    if (step['with'].hasOwnProperty('output_location')) {
+                                        buildConfigs.set('output_location', step['with']['output_location'])
+                                    } else if (step['with'].hasOwnProperty('app_artifact_location')) {
+                                        buildConfigs.set('app_artifact_location', step['with']['app_artifact_location'])
+                                    } else {
+                                        void ext.ui.showWarningMessage(localize('mustContainOutputLocs', `"{0}" must include "output_location" or "app_artifact_location". See the [workflow file guide](https://aka.ms/AAbrcox).`, yamlFile));
+                                        continue;
+                                    }
+                                }
+                            }
+                        }
+                        /* eslint-enable @typescript-eslint/no-unsafe-call, @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access, no-prototype-builtins */
+
+                        treeItems.push(new GitHubConfigGroupTreeItem(parent, yamlFilePath, buildConfigs));
+                    }
+                }
+            }
+
+            return treeItems;
+        }
+
+        return [];
     }
 
     public get iconPath(): TreeItemIconPath {
-        return treeUtils.getThemedIconPath('settings');
+        return new ThemeIcon('settings-gear');
+    }
+
+    public get description(): string {
+        return basename(this.yamlFilePath);
     }
 
     public hasMoreChildrenImpl(): boolean {
         return false;
     }
 
     public async loadMoreChildrenImpl(): Promise<AzExtTreeItem[]> {
-        const buildSettings: string[] = ['app_location', 'api_location', 'app_artifact_location'];
-        return await this.createTreeItemsWithErrorHandling(
-            buildSettings,
-            'azureStaticConfigInvalid',
-            (setting: string) => new ConfigTreeItem(this, setting),
-            (setting: string) => setting
-        );
+        const treeItems: GitHubConfigTreeItem[] = [];
+        this.buildConfigs.forEach((value: string, config: BuildConfig) => {
+            treeItems.push(new GitHubConfigTreeItem(this, config, value));
+        });
+        return treeItems;
```

Comment:
This isn't exactly what I had in mind to resolve the problem.  Now if there's an error, the entire environment node doesn't load.

![image](https://user-images.githubusercontent.com/5290572/113359636-08fb4c00-92fd-11eb-9ef8-ae6b5dc5cfc0.png)

This line is causing the problem: `const yamlFiles: string[] = await readdir(workflowsDir);`

You should be ensuring that this file exists before you try to open it.